main/src/library/Region.flix

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2023 Paul Butcher
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Region {
+
+  ///
+  /// Add x to a queue of `Closeable` to be closed when the region exits:
+  ///
+  /// * `Closeable`s are guaranteed to be closed after any threads within the region have been joined.
+  /// * `Closeable`s are not guaranteed to be closed in any particular order.
+  ///
+  pub def closeOnExit(x: a, r: Region[r]): Unit \ r with Closeable[a] = 
```

Comment:
Per above, if Closable.close is IO then this would need to change too.

Reply:
Is that true? All this is doing is registering the `Closable` to be closed when the region exits. `closeOnExit` has no effect other than to write to the region, surely?

Reply:
Ah right. The effect comes from the `onExit` call, I suppose.

Maybe I'll understand this all better upon seeing some example programs.